runtime/lua/vim/lsp.lua

```diff
@@ -4,15 +4,15 @@ local lsp_rpc = require('vim.lsp.rpc')
 local protocol = require('vim.lsp.protocol')
 local util = require('vim.lsp.util')
 local sync = require('vim.lsp.sync')
-local api = vim.api
 
 local vim = vim
```

Comment:
I think that should be removed; I don't think it's useful except for shutting up linters, which is better (and already) done via linter config. 

(Same in other files.)

Reply:
Reading more about it, it's an optimization technique, since local variable lookup is faster (via hash table) than global variable lookup (via array table). (See http://lua-users.org/wiki/OptimisingUsingLocalVariables)

But I don't know whether this
1. is true for LuaJIT as well
2. makes a measurable difference if so (given that most functions are accessed through lower-tier locals, anyway).

(If 1 and 2 are true, we should do that everywhere, though.)